File: easy/2363_merge_similar_items/solution.cpp
#include <map>
#include <vector>

class Solution {
public:
  std::vector<std::vector<int>>
  mergeSimilarItems(std::vector<std::vector<int>> &items1,
                    std::vector<std::vector<int>> &items2) {
    std::map<int, int> itemsMap;

    for (const std::vector<int> &item : items1)
      itemsMap[item[0]] += item[1];
    for (const std::vector<int> &item : items2)
      itemsMap[item[0]] += item[1];

    std::vector<std::vector<int>> mergedItems;
    mergedItems.reserve(itemsMap.size());

    for (const auto &[value, weight] : itemsMap)
      mergedItems.push_back(std::vector<int>{value, weight});

    return mergedItems;
  }
};

File: easy/2363_merge_similar_items/solution.py
from typing import List


class Solution:
    def mergeSimilarItems(
        self, items1: List[List[int]], items2: List[List[int]]
    ) -> List[List[int]]:
        items_map = {}

        for value, weight in [*items1, *items2]:
            items_map[value] = items_map.get(value, 0) + weight

        return sorted(map(list, items_map.items()))
